lib/finsta_web/live/post_live/index.ex

```diff
@@ -31,15 +31,9 @@ defmodule FinstaWeb.PostLive.Index do
   defp apply_action(socket, :edit, %{"id" => id}) do
     post = Posts.get_post!(id)
```

Comment:
I think we have a problem here now.. an non-owner user is able to see the post data in the edit form, right?

Reply:
The solution would be to pass the user id to `Posts.get_post!`, so it can filter the result scoped to the owner.

Reply:
It could happen if the user realizes you can go to `/posts/49/edit` being 49 a post id that doesn't belong to you.

In this case it woudn't be that dangerous since the data we display in the form is the same we have in the show view. And the user won't be able to submit the form.


Anyways, I agree this is not a good practice and raising a `not found` error will fix the issue (I think Phoenix will render a 404 page)

Reply:
@jmbejar @mgutierrezerpi  So I still need the `Posts.get_post!/1` because we don't need to check the user to show the posts.

What do you think it's better?

- Define a new function `Post.get_post_for_user!(post_id, user_id)`
- Change the actual function to receive options `Post.get_posts!(post_id, opts \\ [])` that we could call like this: `Post.get_posts!(123, user: user)`

Reply:
I'm thinking about what we want to do when the post is not found or belongs to someone else here.

Maybe we should restore the `redirect(to: ~p"/posts")` under both conditions?

Given that, we would have `Post.get_user_post!(user, post_id)` something like this

```ex
post = Post.get_user_post!(user, post_id)

if post do
       socket
       |> assign(:page_title, "Edit Post")
       |> assign(:post, post)
else
       socket
       |> put_flash(:error, "You are not authorized to edit this post")
       |> redirect(to: ~p"/posts")
end
```